test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -748,25 +748,38 @@ private static void deleteAllNodeShutdownMetadata() throws IOException {
     }
 
     protected static void wipeAllIndices() throws IOException {
```

Comment:
It's interesting that there is now a call to the feature reset API in a method that claims to wipe _indices_, because feature reset does more.

I am not suggesting that you tackle this in this PR, but maybe it's worth adding a TODO to revisit cleanup in between tests once we reach a version where feature reset is reliably available.

I think we have too many options for preserve this/preserve that between tests.  I suspect in reality only a small number of combinations of these preservation options are ever set, likely preserve everything (for upgrade tests) and preserve nothing (for almost all other tests).

Calling the feature reset API in this general method that always gets called highlights the inconsistencies.  For example, there's a flag for whether to preserve rollups between tests, but if the feature reset API removed rollups (as it logically should because it removes ML jobs) then calling the feature reset API would render that flag useless.

There are already inconsistencies in this area, because we already call the feature reset API for ML cleanup after ML tests.  There is a comment on that bit of code saying it will wipe more than just ML but it shouldn't matter because everything will be getting wiped between tests anyway.  Since that didn't cause any problems it shows that there's no test suite where both ML tests and non-ML tests run where the fine-grained options to preserve some things but not others are set.

After this PR is merged the cleanup sequence for ML tests will call the feature reset API twice, once to clean up ML, then later when wiping all indices.  That shouldn't break anything because the second call should be a no-op.

Anyway, like I said, don't change any code in this PR because of these issues, but it seems like it would be a good idea to add a TODO and a long term roadmap item to rationalise and simplify cleanup in between tests once feature reset is guaranteed to be available.

Reply:
> it would be a good idea to add a TODO and a long term roadmap item to rationalise and simplify cleanup in between tests once feature reset is guaranteed to be available.

I suggest we do the above and make sure a GH issue is mentioned in the TODO for cleaning this up to help track the origin of the TODO

Reply:
While cleaning this up for merge, it made sense to me to move the feature cleanup to the top level. I'll move it back if this causes any integration test problems.